baselines/datasets/gasdb.py

```diff
@@ -0,0 +1,266 @@
+'''
+Note that much of this code was taken from
+`https://github.com/ulissigroup/cgcnn/`, which in turn was based on
+`https://github.com/txie-93/cgcnn`.
+'''
+
+import os
+from itertools import product
+import numpy as np
+import torch
+from torch_geometric.data import InMemoryDataset, Data
+import ase.db
+from pymatgen.io.ase import AseAtomsAdaptor
+from ..common.registry import registry
+from .base import BaseDataset
+from .elemental_embeddings import EMBEDDINGS
+
+# Import the correct TQDM depending on where we are
+try:
+    shell = get_ipython().__class__.__name__
+    if shell == 'ZMQInteractiveShell':
+        from tqdm.notebook import tqdm
+    else:
+        from tqdm import tqdm
+except NameError:
+    from tqdm import tqdm
+
+
+@registry.register_dataset("gasdb")
+class Gasdb(BaseDataset):
+    def __init__(self, config, transform=None, pre_transform=None):
+        super(BaseDataset, self).__init__(
+            config, transform, pre_transform
+        )
+
+        self.config = config
+        ase_db_path = self.raw_file_names[0]
+        self.ase_db = ase.db.connect(ase_db_path)
+
+        try:
+            self.data, self.slices = torch.load(self.processed_file_names[0])
+            print("### Loaded preprocessed data from:  {}".format(self.processed_file_names))
+        except FileNotFoundError:
+            self.process()
+
+    @property
+    def raw_file_names(self):
```

Comment:
Could we add a note here that it'll only return the first file it finds in the directory.

Reply:
Is it a node to the code-reader or the user? I'm ok with either. We do have a notes to the user in lines 42 and 58.

Reply:
Missed that, sorry. Looks good!